software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcessImpl.java

```diff
@@ -440,12 +441,20 @@ public void rebind() {
         //Only if the expected state is ON_FIRE then the entity has permanently failed.
         Transition expectedState = getAttribute(SERVICE_STATE_EXPECTED);
         if (expectedState == null || expectedState.getState() != Lifecycle.RUNNING) {
+            //FIXME does not set on fire root application
+            if (expectedState.getState() == Lifecycle.STARTING || expectedState.getState() == Lifecycle.STOPPING) {
+                setAttribute(SERVICE_STATE_EXPECTED, new Lifecycle.Transition(Lifecycle.ON_FIRE, new Date()));
+                setAttribute(SERVICE_STATE_ACTUAL, Lifecycle.ON_FIRE);
```

Comment:
Use `ServiceStateLogic.setExpectedState` instead of setting the two attributes yourself.